File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/httpdir/Path.java
package app.zxtune.fs.httpdir;

import android.net.Uri;

import androidx.annotation.Nullable;

public interface Path {
  Uri[] getRemoteUris();
  String getLocalId();
  Uri getUri();
  String getName();
  @Nullable
  Path getParent();
  Path getChild(String name);
  boolean isEmpty();
  boolean isFile();
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/httpdir/Path.kt
package app.zxtune.fs.httpdir

import android.net.Uri

interface Path {
    fun getRemoteUris(): Array<Uri>
    fun getLocalId(): String
    fun getUri(): Uri
    fun getName(): String
    fun getParent(): Path?
    fun getChild(name: String): Path
    fun isEmpty(): Boolean
    fun isFile(): Boolean
}
